test/AutoDiff/stdlib/tgmath_derivatives.swift.gyb

```diff
@@ -114,6 +137,43 @@ DerivativeTests.test("${op}_${T}") {
 %end
     }
   }
+
+  // pow
+  let eps:${T} = 0.01
+  let ulps:${T} = eps/eps.ulp
+  for a in -3...3 {
+    let x = ${T}(a)
+    for b in -3...3 {
+      let y = ${T}(b)
+      var expectedValues: (dx: ${T}, dy: ${T})?
+      if x.isZero {
+        if y.isLess(than: 0) {
+          expectedValues = (dx: ${T}.infinity, dy: ${T}.nan)
+        } else if y.isZero {
+          expectedValues = (dx: ${T}.nan, dy: ${T}.zero)
+        } else if !y.isEqual(to: 1) {
+          expectedValues = (dx: ${T}.zero, dy: ${T}.zero)
+        }
+      } else if x.isLess(than: 2) {
+        expectedValues = (dx: y * pow(x, y - 1), ${T}.zero)
```

Comment:
Can the numerical derivative handle the `x == 1` case, so that you only need a special case for the `x < 0` case here?

Reply:
indeed it is the case :), thanks for catching this ... it made the split in 24939a5 into 3 loops a bit more simple.